### PR TITLE
TRIB-59 TRIB-60: Created controller for api/review-decision, corresponding services and database tables

### PR DESCRIPTION
**Pull request by maryamkhan14**

### CODE REVIEW

File: src/main/java/com/savvato/tribeapp/entities/ReviewDecision.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package com.savvato.tribeapp.entities;
+
+import javax.persistence.*;
+import java.util.Calendar;
+
+@Entity
+@IdClass(ReviewDecisionId.class)
+public class ReviewDecision {
+    @Id
+    public Long reviewId;
```
Comment (haxwell) [inline src/main/java/com/savvato/tribeapp/entities/ReviewDecision.java:10]: The variables can be private. The getters and setters are public.

File: src/main/java/com/savvato/tribeapp/entities/ReviewDecision.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    private java.sql.Timestamp created;
+
+
+    public ReviewDecision(Long reviewId, Long userId, Long reasonId) {
+        this.reviewId = reviewId;
+        this.userId = userId;
+        this.reasonId = reasonId;
+        setCreated();
+    }
+
```
Comment (haxwell) [inline src/main/java/com/savvato/tribeapp/entities/ReviewDecision.java:23]: Add a default (no-parameter) constructor, too.

File: src/main/java/com/savvato/tribeapp/entities/ReviewDecisionReason.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public String getReason() {
+        return reason;
+    }
+
+    public void setReason(String reason) {
+        this.reason = reason;
+    }
+
+    private String reason;
+}
```
Comment (haxwell) [inline src/main/java/com/savvato/tribeapp/entities/ReviewDecisionReason.java:29]: Add an all-parameter, and no-parameter constructor.

File: src/test/java/com/savvato/tribeapp/services/ReviewDecisionServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    @Autowired
+    ReviewDecisionService reviewDecisionService;
+
+    @MockBean
+    ReviewDecisionRepository reviewDecisionRepository;
+    @MockBean
+    ReviewDecisionReasonRepository reviewDecisionReasonRepository;
+
+    // TODO: Confirm no need to test for reviewerId, reviewId being invalid since those should be automatically generated
+    // TODO: Confirm length of test names
```
Comment (haxwell) [inline src/test/java/com/savvato/tribeapp/services/ReviewDecisionServiceImplTest.java:47]: Re: length of test names: Err on the side of being too long.. be verbose.

File: src/test/java/com/savvato/tribeapp/services/ReviewDecisionServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        assertEquals(saveResult.userId, decision.userId);
+        assertEquals(saveResult.reasonId, decision.reasonId);
+    }
+
+    @Test()
+    public void saveReviewDecisionIncorrect() throws Exception {
+        ReviewDecision decision = new ReviewDecision(-1L, 100L, -2L);
+        Mockito.when(reviewDecisionRepository.save(Mockito.any())).thenThrow(new DataAccessException("The save function failed."){});
+
+        assertThrows(DataAccessException.class, ()-> {
```
Comment (haxwell) [inline src/test/java/com/savvato/tribeapp/services/ReviewDecisionServiceImplTest.java:74]: I don't know that this test adds a lot of value.. We're saying, when save() happens, throw an exception. Then we're testing, did an exception happen? That's not really telling us much about the method under test.. 

Let's delete this unhappy path for now. We'll come back to it again as we work to get 100% coverage.

**Review by haxwell — CHANGES_REQUESTED** [review]

File: src/main/java/com/savvato/tribeapp/entities/ReviewDecisionReason.java (hunk 3 above)
Comment (maryamkhan14) [reply src/main/java/com/savvato/tribeapp/entities/ReviewDecisionReason.java:37]: These changes have been added in now.

File: src/test/java/com/savvato/tribeapp/services/ReviewDecisionServiceImplTest.java (hunk 4 above)
Comment (maryamkhan14) [reply src/test/java/com/savvato/tribeapp/services/ReviewDecisionServiceImplTest.java]: Thank you! I have removed this comment.

File: src/main/java/com/savvato/tribeapp/entities/ReviewDecision.java (hunk 2 above)
Comment (maryamkhan14) [reply src/main/java/com/savvato/tribeapp/entities/ReviewDecision.java:26]: I have made this change now.

**Review by haxwell — APPROVED** [review]

**Pull request merged**